### PR TITLE
Framework: remove siteslist usage from lib/cart

### PR DESCRIPTION
**Pull request by gwwar**
This PR removes the SitesList usage from lib/cart and instead provides that information using redux middleware.

### Testing Instructions
- Cart Data should update as expected
- New User Purchases should work
- Existing User purchases should work

### CODE REVIEW

File: client/state/lib/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+Library Middleware
+===========
```
Comment (ockham) [inline client/state/lib/README.md]: Super nitpick-y, but number of `=`s should match no. of chars right above :slightly_smiling_face:

Comment (gwwar) [inline client/state/lib/README.md]: Updated in 879ac00

File: client/state/lib/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	const state = getState();
+	const selectedSite = getSelectedSite( state );
+	library.doSomething( selectedSite );
+```
+
+
+### Use Redux Middleware to setCurrentSite
+
+For libraries that are too large to port, or have too many usages, 
+we can try working around this by setting the selectedSite when it
```
Comment (ockham) [inline client/state/lib/README.md]: The `it` doesn't belong here I think.

File: client/state/lib/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+sites change (eg fetches complete or user sets another site).
+
+First add a new setter function to the library, for example:
+
+```jsx
+library.setSelectedSite( selectedSite );
+```
+
+Then add a handler in this middleware:
+And in this middleware, we can create a handler:
```
Comment (ockham) [inline client/state/lib/README.md]: One of these lines is probably enough :slightly_smiling_face:

File: client/lib/cart/store/index.js [hunk 4]
```diff
@@ -34,6 +33,27 @@ var CartStore = {
 			hasLoadedFromServer: hasLoadedFromServer(),
 			hasPendingServerUpdates: hasPendingServerUpdates()
 		} );
+	},
+	setSelectedSite( selectedSite ) {
```
Comment (ockham) [inline client/lib/cart/store/index.js]: Looks like `selectedSiteId` would suffice for this method.

File: client/state/lib/middleware.js [hunk 5]
```diff
@@ -7,12 +7,25 @@ import { get } from 'lodash';
...
+	SITES_RECEIVE,
+	SITES_UPDATE,
 } from 'state/action-types';
 import analytics from 'lib/analytics';
+import cartStore from 'lib/cart/store';
 import { getSelectedSite } from 'state/ui/selectors';
 import { getCurrentUser } from 'state/current-user/selectors';
 
+/**
+ * Sets the currentSite and siteCount for lib/analytics. This is used to
```
Comment (ockham) [inline client/state/lib/middleware.js]: `s/currentSite/selectedSite, currentUser,/`?

File: client/state/lib/middleware.js [hunk 6]
```diff
@@ -22,14 +35,38 @@ const updateSelectedSiteForAnalytics = ( dispatch, action, getState ) => {
 	analytics.setSiteCount( siteCount );
 };
 
-export const handlers = {
-	[ ANALYTICS_SUPER_PROPS_UPDATE ]: updateSelectedSiteForAnalytics,
+/**
+ * Sets the currentSite for lib/cart/store
```
Comment (ockham) [inline client/state/lib/middleware.js]: `s/currentSite/selectedSite/`

File: client/lib/cart/store/index.js (hunk 4 above)
Comment (gwwar) [inline client/lib/cart/store/index.js]: Good 👀

Comment (gwwar) [inline client/lib/cart/store/index.js]: updated in 2a973be

**Comment by gwwar** [comment]
Thanks for the review @ockham! Let's wait until next week to land, so this give a change for other folks to take a look. 

cc @hoverduck @alisterscott any chance we could run the e2e suite against this branch?

**Comment by gwwar** [comment]
Going to merge to unblock other folks. I'll regression test in all flows on stage.

**Pull request merged**